core/src/test/java/org/apache/iceberg/puffin/TestPuffinWriter.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.puffin;
+
+import java.nio.ByteBuffer;
+import org.apache.iceberg.io.InMemoryOutputFile;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableList;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.junit.Test;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.apache.iceberg.puffin.PuffinCompressionCodec.NONE;
+import static org.apache.iceberg.puffin.PuffinCompressionCodec.ZSTD;
+import static org.apache.iceberg.puffin.PuffinFormatTestUtil.EMPTY_PUFFIN_UNCOMPRESSED_FOOTER_SIZE;
+import static org.apache.iceberg.puffin.PuffinFormatTestUtil.readTestResource;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public class TestPuffinWriter {
+  @Test
+  public void testEmptyFooterCompressed() {
+    InMemoryOutputFile outputFile = new InMemoryOutputFile();
+
+    PuffinWriter writer = Puffin.write(outputFile)
+        .compressFooter()
+        .build();
+    assertThatThrownBy(writer::footerSize)
+        .isInstanceOf(IllegalStateException.class)
+        .hasMessage("Footer not written yet");
+    assertThatThrownBy(writer::finish)
+        .isInstanceOf(UnsupportedOperationException.class)
+        .hasMessage("Unsupported codec: LZ4");
+    assertThatThrownBy(writer::close)
+        .isInstanceOf(UnsupportedOperationException.class)
+        .hasMessage("Unsupported codec: LZ4");
```

Comment:
I'm not sure this is intuitive for users to understand why this exception is happening (given that currently LZ4 is the default when using `compressFooter()` and it's not implemented). Should the default compressor maybe be configured to ZSTD?